File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        unordered_map<int,int> mpp;
        for(int i=0; i<arr.size(); i++){
            mpp[arr[i]]++;
        }
        for(auto it : mpp){
            if(it.second > (0.25*arr.size()))
                return it.first;
        }
        return -1;
    }
};

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
      int size = arr.length;
		int count = 0;
		if (size == 1)
			return arr[0];

		for (int i = 0; i < size - 1; i++) {
			if (arr[i] == arr[i + 1]) {
				count++;
				if (count >= size / 4)
					return arr[i];
			} else {
				count = 0;
				continue;
			}
		}
		return arr[0];   
    }
}